src/__stories__/image-story.tsx

```diff
@@ -1,72 +1,96 @@
+            if: {arg: 'type', eq: 'width and height'},
+        },
+        aspectRatio: {
+            options: ['1 1', '16 9', '7 10', '4 3', '0'],
+            control: {
+                type: 'select',
+                labels: {
+                    '1 1': '1:1',
+                    '16 9': '16:9',
+                    '7 10': '7:10',
+                    '4 3': '4:3',
+                },
+            },
+            if: {arg: 'type', neq: 'width and height'},
+        },
+    },
 };
 
-export const Default: StoryComponent = () => {
-    return (
-        <Stack space={16} dataAttributes={{testid: 'image-story'}}>
-            <Title1>aspectRatio: 1:1</Title1>
-            <Image src={usingVrImg} noBorderRadius aspectRatio="1:1" width={200} />
-            <Title1>aspectRatio: 4:3</Title1>
-            <Image src={usingVrImg} aspectRatio="4:3" width={200} />
-            <Title1>aspectRatio: 7:10</Title1>
-            <Image src={usingVrImg} aspectRatio="7:10" width={200} />
-            <Title1>aspectRatio: 16:9</Title1>
-            <Image src={usingVrImg} aspectRatio="16:9" width={200} />
-            <Title1>width: 230; height: 100</Title1>
-            <Image src={usingVrImg} width={230} height={100} />
-            <div
-                style={{
-                    resize: 'both',
-                    overflow: 'auto',
-                    width: 320,
-                    border: `1px solid ${skinVars.colors.border}`,
-                    padding: 16,
-                }}
-            >
-                <Stack space={16}>
-                    <Title1>width: 50%; aspectRatio: 4:3</Title1>
-                    <Image src={usingVrImg} width="50%" aspectRatio={4 / 3} />
+type Args = {
+    type: 'width and height' | 'width and aspect ratio' | 'full width';
+    width: number;
+    height: number;
+    aspectRatio: string;
+    borderRadius: boolean;
+    isInvalidSource: boolean;
+};
 
-                    {!isRunningAcceptanceTest() && (
-                        // for some reason, adding this image to the screenshot test makes it unstable
-                        <Stack space={16}>
-                            <Title1>width: 50%; aspectRatio: 0</Title1>
-                            <Image src={usingVrImg} width="50%" aspectRatio={0} />
-                        </Stack>
-                    )}
+export const Default: StoryComponent<Args> = ({
+    type,
+    width,
+    height,
+    aspectRatio,
+    borderRadius,
+    isInvalidSource,
+}) => {
+    const props = {
+        width: type !== 'full width' ? width : undefined,
+        height: type === 'width and height' ? height : undefined,
+        aspectRatio:
+            type !== 'width and height'
+                ? aspectRatio === '0'
+                    ? 0
+                    : (aspectRatio.replace(' ', ':') as AspectRatio)
+                : undefined,
+        noBorderRadius: borderRadius === false,
+        dataAttributes: {testid: 'image'},
```

Comment:
see comment in Image component, you could use the `data-component-name` in tests, instead of defining the `testid`

Same in other tests

Reply:
Discussed offline. We decided to keep the data-testid, because we use this id specifically for testing purposes and it makes sense to differentiate it from the component name.